bolt/lib/Target/AArch64/AArch64MCSymbolizer.cpp

```diff
@@ -127,13 +127,37 @@ AArch64MCSymbolizer::adjustRelocation(const Relocation &Rel,
     assert(SymbolValue && "Symbol value should be set");
     const uint64_t SymbolPageAddr = *SymbolValue & ~0xfffULL;
 
-    AdjustedRel.Symbol = BC.registerNameAtAddress("__BOLT_got_zero", 0, 0, 0);
-    AdjustedRel.Addend = Rel.Value;
+    // Check if defined symbol and GOT are on the same page. If they are not,
+    // disambiguate the operand.
+    if (BC.MIB->isADRP(Inst) && Rel.Addend == 0 &&
```

Comment:
Would it make sense to also check that the type here is `R_AARCH64_ADR_GOT_PAGE` ?

Reply:
This code is under `Relocation::isGOT()` condition, so this would be redundant.